### PR TITLE
Add test to ensure no duplicate values in ReactSymbols

### PR DESCRIPTION
**Pull request by bvaughn**
Prevent future clowniness like #12843 🤡

### CODE REVIEW

File: packages/shared/__tests__/ReactSymbols-test.internal.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @emails react-core
+ */
+'use strict';
+
+describe('ReactSymbols', () => {
+  beforeEach(() => jest.resetModules());
+
+  const expectToBeUnique = keyValuePairs => {
+    const map = new Map();
```
Comment (gaearon) [inline packages/shared/__tests__/ReactSymbols-test.internal.js]: Maybe would be easier to get an array with `Object.values()` and then verify its length is the same as `new Set(arr).size`?

File: packages/shared/__tests__/ReactSymbols-test.internal.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      map.set(value, key);
+    });
+  };
+
+  it('Symbol values should be unique', () => {
+    expectToBeUnique(Object.entries(require('shared/ReactSymbols')));
+  });
+
+  it('numeric values should be unique', () => {
+    global.Symbol.for = null;
```
Comment (gaearon) [inline packages/shared/__tests__/ReactSymbols-test.internal.js]: I think you'll want to restore this after the test has run. It's a global so I'd expect it to be shared between tests. At least within one file so the order starts mattering.

```
const originalSymbolFor = global.Symbol.for;
global.Symbol.for = null;
try {
  // ...
} finally {
  global.Symbol.for = originalSymbolFor;
}
```

File: packages/shared/__tests__/ReactSymbols-test.internal.js (hunk 1 above)
Comment (bvaughn) [reply packages/shared/__tests__/ReactSymbols-test.internal.js]: Yeah, that would be easier but this provides a more meaningful error message if it fails. 😄

File: packages/shared/__tests__/ReactSymbols-test.internal.js (hunk 2 above)
Comment (bvaughn) [reply packages/shared/__tests__/ReactSymbols-test.internal.js]: I thought Jest automatically reset modules between test files. 🤔

But yeah, still probably a good suggestion.

Comment (bvaughn) [reply packages/shared/__tests__/ReactSymbols-test.internal.js]: Agreed 👍

**Pull request merged**